docs/fundamentals/code-analysis/style-rules/naming-rules.md

```diff
@@ -19,75 +19,75 @@ helpviewer_keywords:
 ---
 # Code-style naming rules
 
-In your `.editorconfig` file, you can define **naming rules** that specify and enforce how .NET programming language code elements&mdash;such as classes, properties, and methods&mdash;should be named. For example, you can specify that public members must be capitalized, or that private fields must begin with `_`.
+In your `.editorconfig` file, you can define **naming rules** that specify and enforce how .NET programming language code elements &ndash;&nbsp;such as classes, properties, and methods&nbsp;&ndash; should be named. For example, you can specify that public members must be capitalized, or that private fields must begin with `_`.
 
 A naming rule has three components:
 
-* The **symbol group** that the rule applies to, for example, public members or private fields.
-* The **naming style** to associate with the rule, for example, that the name must be capitalized or start with an underscore.
-* The severity for enforcing the convention.
+1. The **symbol group** that the rule applies to; for example, public members or private fields.
+1. The **naming style** to apply with the rule; for example, the name must be capitalized or start with an underscore.
+1. The **naming rule**, linking
```

Comment:
Hi @zspitz,

thank you for taking the time.

> A naming rule is not a component of itself.

Absolutely. I left the original ambiguity here. In fact, the original document uses the same term for both. Though being unsatisfied with the double meaning, I didn't dare creating a new technical term by myself as I didn't want to introduce new technical terms without your consent.

One approach to solve this would be not to intermix all three components into a single page:

If we'd be perfectly correct, the current document actually is sort of misleading by mixing three rather unrelated parts into a single page:

1. There is a ***symbol group*** which may be used by `.editorconfig` components other than naming rules.
1. There is a ***naming style*** which may be used by `.editorconfig` components other than naming rules.
1. There is a ***naming rule***, merely referencing these unrelated other components.

![Naming Rule](https://user-images.githubusercontent.com/9283914/133243206-7d887e0e-367f-4b82-8367-5c3e1ad26931.png)

*Symbol groups*, *naming styles* and *naming rules* are three distinct components, not three parts of one single component. A *naming rule* doesn't "inherit" *symbol groups* and *naming styles*, it rather "references" them.

Another `.editorconfig` components may want to make use of ***symbol groups*** in the future. So it'd maybe make sense to split documentation of these three components into three distinct documents.

This would also solve the ambiguity, because a *naming rule* would just be the "severity" component, referencing two other distinct components.

Reply:
With your approval I would create three distinct documents from this single one.

Reply:
This is a great discussion. These naming rules are not easy to document in a straightforward way.

I'm not sure that we need separate articles for symbol groups and naming styles, since at this point they only apply to naming rules. I propose we take away the abstraction (e.g. this section https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/naming-rules#general-syntax) which I think adds to the confusion. Users trying to create a naming rule don't necessarily need to know that all 3 components conform to the same pattern. We can just describe each component as a separate entity in its own H2 (##) section. How does that sound?

Reply:
> I propose we take away the abstraction

I would agree, except that all 3 element kinds have no specific "element definition"; each definition is actually the sum of the property settings for that element kind and shared title.

It was a source of frustration to me until I understood that in order to create a symbol group called `all_classes` I didn't have to create a section headed `dotnet_naming_symbols.all_classes`, under which I might set the relevant properties specific to that symbol group; but that all I needed to do was to define one (or more) property settings using 1) that element type and 2) title:

   dotnet_naming_symbols.all_classes.applicable_kinds = class

I don't know of many configuration systems that work like this, and I think it sufficiently unusual that we need to call attention to it.

In contrast, an XML-, JSON-, or YAML-based configuration system wouldn't need an explicit guide to their underlying syntax because those syntaxes are standardized and well-known.

Reply:
I'm not sure if taking away the abstraction alone will suffice.

You may have already noticed: I also cleaned up the document structure to reflect the fact that three distinct components are required to create a Naming Rule:

![Sequence](https://user-images.githubusercontent.com/9283914/133943783-89dd052c-7af1-43e3-b45e-126546300b0f.png)

Reply:
> Provided you'd be giving me your approval I would go and create three distinct documents from this current single one.

I think this might be a good way to proceed. I'd like to see something in the TOC like:

- Naming-related options
  - Symbol groups
  - Naming styles
  - Naming rules